lib/node_modules/@stdlib/regexp/to-json/lib/main.js

```diff
@@ -0,0 +1,73 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isRegExp = require( '@stdlib/assert/is-regexp' );
+var format = require( '@stdlib/string/format' );
+
+
+// MAIN //
+
+/**
+* Return a JSON representation of a regexp object.
+*
+* @param {RegExp} regex - regular expression to serialize
+* @throws {TypeError} first argument must be an RegExp object
+* @returns {Object} JSON representation
+*
+* @example
+* var pattern = 'ab+c';
+* var regex = new RegExp( pattern );
+* var json = toJSON( regex );
+* // returns {...}
+*/
+function toJSON( regex ) {
+	var pattern;
+	var out;
+
+	if ( !isRegExp( regex ) ) {
+		throw new TypeError( format( 'invalid argument. Must provide a regular expression. Value: `%s`.', regex ) );
+	}
+	out = {};
+
+	// Guaranteed properties:
+	regex = String(regex).split('/');
+	pattern = regex.slice(1, -1);
+
+	// Insert the missing '/' inside of the pattern
+	if ( pattern.length > 1 ) {
+		pattern = pattern.join( '/' );
+	}
+	else {
+		pattern = pattern[0];
+	}
+
+	out.type = 'RegExp';
+	out.flags = regex[ regex.length - 1 ];
+	out.pattern = pattern;
+
+	return out;
```

Comment:
The approach, as implemented, will work, but I think we can avoid temporary array and string creation and branching by using [`@stdlib/regexp/regexp`](https://github.com/stdlib-js/stdlib/tree/f500d1d5c573fcfe78e774b7895499e1014aa9f9/lib/node_modules/%40stdlib/regexp/regexp).

```suggestion
	var parts;
	var re;
	if ( !isRegExp( regex ) ) {
		throw new TypeError( format( 'invalid argument. Must provide a regular expression. Value: `%s`.', regex ) );
	}
	re = reRegExp();
	parts = re.exec( regex.toString() );
	return {
		'type': 'RegExp',
		'pattern': parts[ 1 ],
		'flags': parts[ 2 ]
	};
```